src/core/swap.c

```diff
@@ -771,10 +771,10 @@ static void swap_enter_activating(Swap *s) {
                 r = fstab_find_pri(s->parameters_fragment.options, &priority);
                 if (r < 0)
                         log_warning_errno(r, "Failed to parse swap priority \"%s\", ignoring: %m", s->parameters_fragment.options);
-                else if (r == 1 && s->parameters_fragment.priority >= 0)
+                else if (r == 1 && s->parameters_fragment.priority != 0)
```

Comment:
This does not work. If priority can be positive and negative, I'm pretty sure it can be 0 too.
So we probably need to bother with `s->parameters_fragment.priority_set` or something ugly like that.

Reply:
Or are we treating pri=0  as unset?

Reply:
That was my thinking. But yes, with `s->parameters_fragment.priority_set` we could reject `"pri=0,pri=0"`, `"pri=0,pri=1"` etc.